mesonbuild/backend/ninjabackend.py

```diff
@@ -1838,7 +1838,10 @@ def generate_cython_transpile(self, target: build.BuildTarget) -> \
 
         for src in target.get_sources():
             if src.endswith('.pyx'):
-                output = os.path.join(self.get_target_private_dir(target), f'{src}.{ext}')
+                # Use basename to avoid too nested targets which can cause a
+                # problem with MAX_PATH on Windows
+                basename = src.fname
```

Comment:
fname can have path seperators, it's the path relative to the directory that the `File` object is defined in, so if the src is `files('foo/bar/x.pyx')`, then fname will reflect that.